hooks/scheduler.js

```diff
@@ -92,11 +97,12 @@ class Scheduler {
 
   /** Handle spawning a new task for a given hook that needs to be scheduled */
   async handleHook(hook) {
-    let task = await hook.taskPayload();
-    let resp = await queue.createTask(hook.nextTaskId, task);
+    // TODO: (when we have hook logging) if this fails due to 401, we should
+    // still consider it scheduled
+    await this.taskcreator.fire(hook, {}, {taskId: hook.nextTaskId});
     await hook.modify((hook) => {
-      hook.nextTaskId        = slugid();
-      hook.nextScheduledDate = utils.nextDate(hook.schedule);
+      hook.nextTaskId        = slugid.v4();
+      hook.nextScheduledDate = nextDate(hook.schedule);
```

Comment:
Use `taskcluster.slugid()` (gives you a nice one... and then everybody uses the same method for taskIds)...

More notably... I'm wrondering if it wouldn't be sane to do:
```js
let taskId = hook.nextTaskId;
hook.modify(hook => {
  if (taskId === hook.nextTaskId) {
    hook.nextTaskId = taskcluster.slugid();
    hook.nextScheduledDate = ....
  }
});
```
I guess it's not critical... but if two schedulers are racing (a real senario), this would have some effect if the next time schedule extremely close... or there is some clock drift.. otherwise, this wouldn't matter and is just a sanity thing...